Replace Visual Studio with MinGW-w64 for Windows platforms

catageek opened the pull request:
According to http://stackoverflow.com/questions/19425482/windows-c-compiler-with-full-c11-support-should-work-with-qt and my own experience, we should stop supporting Microsoft Visual Studio on this project for various reasons:
- Many C++11 features are not implemented (constexpr, noexcept,...)
- Some features have bugs that are not yet fixed (std::chrone::steady_clock)
- There are certainly other bugs not yet found

As a logical result, a code that is fully compliant with C++11 and builds perfectly with gcc must be modified, reimplemented or debugged to be compatible with Visual Studio.

The impact is that some coders have to make some additional work on the code to make it compatible, and this adds a delay to each PR that Visual Studio cannot build (currently 2 weeks on the component framework which is needed to step forward).

So:
- unless we consider that we have so many coders that we can make them work a a non-essential task,
- unless we have so much time to spill to make the code compatible with a non-free tool,
- unless we want spend 8 years on this project because we can't force Windows people to fix the code in time,

we should stop supporting the product to spend some time on the code of the game itself.

Commit:
Fix previous merge

Commit:
Fix bug in bitmap bool operations

Commit:
Remove 'constexpr' since VS2013 does not support it

Commit:
Add a function to get a non-const copy of the data

Commit:
Merge branch 'develop' of github.com:adam4813/trillek-client-core into
feature-component-framework

Commit:
Update assets to include volocity components

Commit:
Add information on JSON parse error

Commit:
Remove libkqueue and netport dependency, to be replaced by raknet

Commit:
Update to last assets commit

adam4813 committed:
Initial commit for VComputerSystem

There is currently only 1 keybaord available for all devices, but
each device can have it's own screen and associated output
PixelBuffer.

adam4813 committed:
Actually added the vcomputer-module

Commit:
Add RakNet library as dependency

Commit:
Merge branch 'feature-cpu' of https://github.com/adam4813/trillek-client-core into feature-component-framework

Conflicts:
	.gitmodules
	CMakeLists.txt
	assets

Commit:
Add ignore dirty to submodules to avoid extra modification status

Commit:
Fix build errors in previous commit

Commit:
Fix crypto++ submodule

Commit:
Merge branch 'develop' of https://github.com/adam4813/trillek-client-core into feature-raknet

Commit:
Fix crypto++ submodule declaration

Commit:
Fix vcomputer library variable in CMakeList.txt

Commit:
Update to assets removing double addition of components

Commit:
Add Bullet to appveyor script

adam4813 commented:
The newest Visual C++ (due out in a few months) should add in many of the missing features that we will need.

Also, as someone who tried before, MinGW is neither a good drop-in, nor is is painless by any means to use.

Commit:
Use C++ operators new/delete for memory allocations

Commit:
Fix memory leak in Collidable

Commit:
Align VelocityStruct on a 16-byte boundary

The macros used are also used in btVector3 to be sure that the
proper alignment is applied.

adam4813 committed:
Some minor typos and Visual C++ oddities
Typos:
-frame_tp_tp instead of just frame_tp
- A malformed target name in CMakeLists (tests build section)

The oddities include:
-std::min must be surrounded in () to enforce
the std function instead of the min macro.
-Removing the last references of constexpr
-Visual C++ declares std::make_unique and so we must complement
the ifdef for non msvc with a using directive.
-__popcnt64 isn't available unless doing a 64bit build
-std::vector initialization lists in construction initializer lists isn't allowed

Errors that need resolving:
-transform.cpp glm::eualarAnglers conversion issues
-Visual C++ friend template methods with fully qualified
namespaces causes an internal compiler error, so removing
the friend reference requires make several classes private
data public for access level reasons.
-Registering a Shared component in registration fails with
internal compiler?!?!
-Custom allocator can't align 16 for VelocityStruct when using
btVector3

catageek committed:
Merge pull request #4 from adam4813/feature-component-framework

Some minor typos and Visual C++ oddities

Commit:
Merge branch 'feature-raknet' into feature-component-framework

Conflicts:
	.gitmodules

Commit:
Remove network code and crypto++ library

Commit:
Fix minor errors in vcomputer system

Commit:
Fix travis for GLFW since GLFW now wants cmake-2.8.9

Commit:
Remove network and crypto tests

Commit:
Update vcomputer submodule

Commit:
Update vcomputer submodule

The submodule had a build failing typo only visible
when you had multiple projects trying to add tests
and build/find gtest.

Commit:
Use explicit constructor of glm::vec3

Zardoz89 committed:
Now gets GLM 9.5.1 from github

Zardoz89 committed:
Fixed unzip file name

Zardoz89 committed:
Same version that uses appveyor to check Window building

Commit:
Fix signature of Initialize() for ReferenceFrame component

adam4813 committed:
Resolved some Visual C++ issues with templates

adam4813 committed:
Fixed Transform::SetRotation and orientation

The eular angles are computed manually (based on the
function in glm::eularAngeles)

On Visual C++ glm::atan2 for float isn't being found, but using
atan2 from cmath works just fine (glm calls through to this anyways).

adam4813 committed:
Merge remote-tracking branch 'catageek/feature-component-framework' into feature-component-framework

adam4813 committed:
Merge branch 'feature-component-framework' into develop

Conflicts:
	.travis.yml
	CMakeLists.txt
	include/graphics/camera.hpp
	include/graphics/six-dof-camera.hpp
	include/physics/collidable.hpp
	include/trillek-game.hpp
	src/systems/physics.cpp
	src/trillek-game.cpp
